src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -187,20 +187,27 @@ public Image<TPixel> Decode<TPixel>(BufferedReadStream stream, CancellationToken
             where TPixel : unmanaged, IPixel<TPixel>
         {
             using var spectralConverter = new SpectralConverter<TPixel>(this.Configuration);
-
-            var scanDecoder = new HuffmanScanDecoder(stream, spectralConverter, cancellationToken);
-
-            this.ParseStream(stream, scanDecoder, cancellationToken);
-            this.InitExifProfile();
-            this.InitIccProfile();
-            this.InitIptcProfile();
-            this.InitXmpProfile();
-            this.InitDerivedMetadataProperties();
-
-            return new Image<TPixel>(
-                this.Configuration,
-                spectralConverter.GetPixelBuffer(cancellationToken),
-                this.Metadata);
+            try
+            {
+                var scanDecoder = new HuffmanScanDecoder(stream, spectralConverter, cancellationToken);
+
+                this.ParseStream(stream, scanDecoder, cancellationToken);
+                this.InitExifProfile();
+                this.InitIccProfile();
+                this.InitIptcProfile();
+                this.InitXmpProfile();
+                this.InitDerivedMetadataProperties();
+
+                return new Image<TPixel>(
+                    this.Configuration,
+                    spectralConverter.GetPixelBuffer(cancellationToken),
+                    this.Metadata);
+            }
+            catch
+            {
+                this.Frame?.Dispose();
+                throw;
+            }
```

Comment:
There's no need to do a try-catch here I believe as it is handled by the caller here:

https://github.com/SixLabors/ImageSharp/blob/136cc143bba2fca255328751dd4580e4f8d413c3/src/ImageSharp/Formats/Jpeg/JpegDecoder.cs#L19-L26

Reply:
yeah looks like you are right... I think I must have added it on first pass trying to find the leaky code.

Reply:
It's very tricky because jpeg is the only decoder which has a disposable internal `DecoderCore` implementation :)

Reply:
Should we make it so it’s doesn’t implement IDisposable?

Reply:
Hmm... Considering it's only used for `Frame` field which is a class field for the sake of not passing it to a bunch of methods we can make it non-disposable for consistency.

If you do so, try-catch must be replaced by try-finally or `Frame` would leak in a no-exception scenario. Also don't forget about `Identify` method - it also creates a `Frame` field so it should have a proper try-finally too.

Reply:
Thanks. Ace, that’s what I thought. That can be a separate PR though.